symengine/functions.cpp

```diff
@@ -2895,6 +2895,108 @@ RCP<const Basic> uppergamma(const RCP<const Basic> &s, const RCP<const Basic> &x
 }
 
 
+PolyGamma::PolyGamma(const RCP<const Basic> &n, const RCP<const Basic> &x)
+    : n_{n}, x_{x}
+{
+    SYMENGINE_ASSERT(is_canonical(n_, x_))
+}
+
+bool PolyGamma::is_canonical(const RCP<const Basic> &n, const RCP<const Basic> &x)
+{
+    if(is_a<Integer>(*n)) {
+        auto n_int = rcp_static_cast<const Integer>(n); 
+        if(n_int->is_negative()) {
+            return false;
```

Comment:
`is_canonical` should return false only when that object is not created from the method `polygamma` which means `is_canonical` is a check to prevent users from creating invalid objects by using the constructor directly. For example, `Polygamma(0, 1)` can never be there if `polygamma(0, 1)` is called. (Note the lowercase and uppercase)